app/src/test/contrib/actions/DemographicActionsTests.ts

```diff
@@ -0,0 +1,92 @@
+import { expect } from "chai";
+
+import { Sandbox } from "../../Sandbox";
+import { demographicActionCreators } from "../../../main/contrib/actions/demographicActionCreators";
+import { DemographicService } from "../../../main/contrib/services/DemographicService";
+import { DemographicTypes } from "../../../main/contrib/actionTypes/DemographicTypes";
+import {createMockStore} from "../../mocks/mockStore";
+import {mockDemographicDataset, mockTouchstone} from "../../mocks/mockModels";
+
+describe("Demographic actions tests", () => {
+    const sandbox = new Sandbox();
+
+    const testDemographicDataSet = mockDemographicDataset({id: "set-1"});
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it("sets fetched sets", (done) => {
+        const store = createMockStore({});
+        sandbox.setStubFunc(DemographicService.prototype, "getDataSetsByTouchstoneId", ()=>{
+            return Promise.resolve([testDemographicDataSet]);
+        });
+        store.dispatch(demographicActionCreators.getDataSets('touchstone-1'))
+        setTimeout(() => {
+            const actions = store.getActions()
+            const expectedPayload = { type: DemographicTypes.DEMOGRAPHIC_DATA_SETS_FETCHED, data: [testDemographicDataSet] }
+            expect(actions).to.eql([expectedPayload])
+            done();
+        });
+    });
+
+    it("set data set by id using previously loaded sets", (done) => {
+        const store = createMockStore({demographic: {dataSets: [testDemographicDataSet] } });
+        store.dispatch(demographicActionCreators.setDataSet('set-1'))
+        setTimeout(() => {
+            const actions = store.getActions()
+            const expectedPayload = { type: DemographicTypes.DEMOGRAPHIC_SET_DATA_SET, data: testDemographicDataSet }
+            expect(actions).to.eql([expectedPayload])
+            done();
+        });
+    });
+
+    it("set gender", (done) => {
+        const store = createMockStore({});
+        store.dispatch(demographicActionCreators.setGender('male'))
+        setTimeout(() => {
+            const actions = store.getActions()
+            const expectedPayload = { type: DemographicTypes.DEMOGRAPHIC_SET_GENDER, data: 'male' }
+            expect(actions).to.eql([expectedPayload])
```

Comment:
There's some boiler plate repeated through these tests. Frequently: 1. get the actions 2. state the expectation 3. check the actions equals singleton array of expected payload. It would be better to just have (as we had previously) a helper along the lines of 

`expectOneAction({ type: DemographicTypes.DEMOGRAPHIC_SET_GENDER, data: 'male' })`

This will remove a fair few identical lines.

Reply:
testing helpers in not a good idea in general, hides logic, makes things obscure